src/marketplace/MakeAComponentSaleView.js

```diff
@@ -0,0 +1,93 @@
+//@flow strict
+import * as React from 'react';
+import Cookies from 'universal-cookie';
+
+type Props = {||}; 
+type State = {|
+    userId: string,
+    salePrice: number,
+    itemID: string,
+    itemAmount: number,
+|};
+let itemsToSell = [{value: '', label: ''}];
+class MakeAComponentSaleView extends React.Component<Props, State> {
+
+    
+    constructor() {
+        super();
+        this.state={
+            userId: '',
+            salePrice: 0,
+            itemID: '',
+            itemAmount: 0,
+        }
+        this.getUserInventory();
+    }
+
+
+    getUserInventory = async ():Promise<void> => {
+        itemsToSell = [];
```

Comment:
This is super unsafe because it runs asynchronously and is modifying stuff that multiple threads might touch.